nbclient/util.py

```diff
@@ -102,3 +103,15 @@ async def ensure_async(obj: Union[Awaitable, Any]) -> Any:
         return result
     # obj doesn't need to be awaited
     return obj
+
+
+def run_hook(hook: Optional[Callable], **kwargs) -> None:
+    if hook is None:
+        return
+    if inspect.iscoroutinefunction(hook):
+        future = hook(**kwargs)
+    else:
+        loop = asyncio.get_event_loop()
+        hook_with_kwargs = partial(hook, **kwargs)
+        future = loop.run_in_executor(None, hook_with_kwargs)
+    asyncio.ensure_future(future)
```

Comment:
I don't understand, does it mean that hooks are going to be launched in the background? BTW, it would be nice to test async hooks.
I think `run_hook` should be `async`:
```python
async def run_hook(hook: Optional[Callable], **kwargs) -> None:
    await ensure_async(hook(**kwargs))
```

Reply:
Yeah, the intent is for the hook to be launched in the background.